Fix issue clicking on 2D view in Slice Viewer

jclarkeSTFC opened the pull request:
### Description of work
If a peaks workspace has been deleted then the corresponding presenter in Slice Viewer won't exist. If the `Add Peak` button has been selected then if you click on the plot it will try and call methods on the missing presenter.

Fixes #36266. See that same issue for steps to reproduce. <!-- and fix #xxxx or close #xxxx xor resolves #xxxx. One line per issue fixed. -->
<!-- alternative
*There is no associated issue.*
-->

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

### To test:
Adding/removing peaks in Slice Viewer is the thing most likely to go wrong.

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

jclarkeSTFC committed:
Check for valid peak presenter

The peaks workspace may have been deleted, and hence the presenter, hence
double-check before trying to add or remove any more peaks.

jclarkeSTFC committed:
Added release notes

jhaigh0 reviewed (dismissed):
Simple fix which works well 👍

jclarkeSTFC committed:
Add test to check behaviour is fixed

The new test will remove the only peaks workspace, then attempt to add a
new peak. Previously this would result in an exception being thrown.

jclarkeSTFC committed:
Removed redundant pass

jhaigh0 reviewed (approved):
New test looks good, approved providing the test passes
